ospfd/ospf_vty.c

```diff
@@ -5711,27 +5716,61 @@ DEFUN (show_ip_ospf_instance_neighbor_int_detail,
 }
 
 /* Show functions */
-static int show_lsa_summary(struct vty *vty, struct ospf_lsa *lsa, int self)
+static int show_lsa_summary(struct vty *vty, struct ospf_lsa *lsa, int self,
+			    json_object *json_lsa)
 {
 	struct router_lsa *rl;
 	struct summary_lsa *sl;
 	struct as_external_lsa *asel;
 	struct prefix_ipv4 p;
+	char buf[PREFIX2STR_BUFFER];
 
 	if (lsa != NULL)
 		/* If self option is set, check LSA self flag. */
 		if (self == 0 || IS_LSA_SELF(lsa)) {
-			/* LSA common part show. */
-			vty_out(vty, "%-15s ", inet_ntoa(lsa->data->id));
-			vty_out(vty, "%-15s %4d 0x%08lx 0x%04x",
-				inet_ntoa(lsa->data->adv_router), LS_AGE(lsa),
-				(unsigned long)ntohl(lsa->data->ls_seqnum),
-				ntohs(lsa->data->checksum));
+
+			if (!json_lsa) {
+				/* LSA common part show. */
+				vty_out(vty, "%-15s ",
+					inet_ntoa(lsa->data->id));
+				vty_out(vty, "%-15s %4d 0x%08lx 0x%04x",
+					inet_ntoa(lsa->data->adv_router),
+					LS_AGE(lsa),
+					(unsigned long)ntohl(
+						lsa->data->ls_seqnum),
+					ntohs(lsa->data->checksum));
+			} else {
+				char seqnum[10];
+				char checksum[10];
+
+				snprintf(seqnum, 10, "%x",
```

Comment:
Would you mind to change the hard coded size of `10` to `sizeof(seqnum)`? Same applies for the next statement with `checksum`.

Reply:
Addressed